### DISCUSSION TITLE
vim: add unique sort

### DISCUSSION THREAD

**Discussion by hugginsio** [zed-industries/zed#21309] [Feature Requests]
Zed combines Vim mode with the built-in Command Palette. [Many common Vim commands are aliased to their Zed-specific equivalents][1]. Among these are `sort` and `sort i`:

<img width="487" alt="image" src="https://github.com/user-attachments/assets/762f0c0b-6570-4e4a-95f3-832ec2d0b436">

Another common shortcut is `sort u`, used to sort and remove duplicate lines. Zed already has a built-in that can take care of the latter, `editor::UniqueLinesCaseInsensitive`.

Would the team support the addition of another Vim alias to support `sort u`, which would combine `editor::SortLinesCaseSensitive` and `editor::UniqueLinesCaseInsensitive`?

[1]: https://zed.dev/docs/vim#command-palette

**Comment by hugginsio** [comment]
Looks like changes for this would probably start around `crates/vim/src/command.rs`:

https://github.com/zed-industries/zed/blob/eb2c0b33dff361a268b0276f8ec8bb38811d2c5e/crates/vim/src/command.rs#L674-L675

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
